flink-ml-lib/src/test/java/org/apache/flink/ml/classification/knn/KnnTest.java

```diff
@@ -0,0 +1,273 @@
+import org.apache.commons.collections.IteratorUtils;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+import static org.junit.Assert.assertEquals;
+
+/** Tests Knn and KnnModel. */
+public class KnnTest {
+    private StreamExecutionEnvironment env;
+    private StreamTableEnvironment tEnv;
+    private Table trainData;
+    private static final String LABEL_COL = "test_label";
+    private static final String PRED_COL = "test_prediction";
+    private static final String VEC_COL = "test_features";
+    private static final List<Row> trainArray =
+            new ArrayList<>(
+                    Arrays.asList(
+                            Row.of(1, Vectors.dense(2.0, 3.0)),
+                            Row.of(1, Vectors.dense(2.1, 3.1)),
+                            Row.of(2, Vectors.dense(200.1, 300.1)),
+                            Row.of(2, Vectors.dense(200.2, 300.2)),
+                            Row.of(2, Vectors.dense(200.3, 300.3)),
+                            Row.of(2, Vectors.dense(200.4, 300.4)),
+                            Row.of(2, Vectors.dense(200.4, 300.4)),
+                            Row.of(2, Vectors.dense(200.6, 300.6)),
+                            Row.of(1, Vectors.dense(2.1, 3.1)),
+                            Row.of(1, Vectors.dense(2.1, 3.1)),
+                            Row.of(1, Vectors.dense(2.1, 3.1)),
+                            Row.of(1, Vectors.dense(2.1, 3.1)),
+                            Row.of(1, Vectors.dense(2.3, 3.2)),
+                            Row.of(1, Vectors.dense(2.3, 3.2)),
+                            Row.of(3, Vectors.dense(2.8, 3.2)),
+                            Row.of(4, Vectors.dense(300., 3.2)),
+                            Row.of(1, Vectors.dense(2.2, 3.2)),
+                            Row.of(5, Vectors.dense(2.4, 3.2)),
+                            Row.of(5, Vectors.dense(2.5, 3.2)),
+                            Row.of(5, Vectors.dense(2.5, 3.2)),
+                            Row.of(1, Vectors.dense(2.1, 3.1))));
+
+    private static final List<Row> testArray =
+            new ArrayList<>(
+                    Arrays.asList(
+                            Row.of(5, Vectors.dense(4.0, 4.1)), Row.of(2, Vectors.dense(300, 42))));
+    private Table testData;
+
+    @Before
+    public void before() {
+        Configuration config = new Configuration();
+        config.set(ExecutionCheckpointingOptions.ENABLE_CHECKPOINTS_AFTER_TASKS_FINISH, true);
+        env = StreamExecutionEnvironment.getExecutionEnvironment(config);
+        env.setParallelism(4);
+        env.enableCheckpointing(100);
+        env.setRestartStrategy(RestartStrategies.noRestart());
+        tEnv = StreamTableEnvironment.create(env);
+
+        Schema schema =
+                Schema.newBuilder()
+                        .column("f0", DataTypes.INT())
```

Comment:
Previously I also think it is cleaner to use `int` to represent categorical field. And `KMeans` currently outputs the prediction column as `int`.

However, @zhipeng93 said most algorithm users prefer the output/input type to be consistently `double`. And Spark also use `double` to represent categorical field.

While I don't have a strong opinion on this issue, IMO it is probably useful to follow the same practice in Flink ML.

Could you check with @zhipeng93 and decide whether we should use int or double here?

Reply:
After I discuss with @zhipeng93, I decide using double to be the label type.

Reply:
Sounds good.